src/ripple/app/paths/impl/XRPEndpointStep.cpp

```diff
@@ -249,6 +263,13 @@ XRPEndpointStep::check (StrandContext const& ctx) const
         return temBAD_PATH;
     }
 
+    // If we're offer crossing, only fiddle with the reserve if a trust line
+    // from acc_ to the strand's delivered currency does not yet exist.
+    if (reduceReserve_)
```

Comment:
I don't like `check` setting state. I can see why you did it, however, I'd rather pass the `view` into the constructor than have `check` change state.